dexmaker/src/main/java/com/google/dexmaker/AppDataDirGuesser.java

```diff
@@ -61,12 +61,55 @@ private String getPathFromThisClassLoader(ClassLoader classLoader,
         }
 
         // Parsing toString() method: yuck.  But no other way to get the path.
-        // Strip out the bit between square brackets, that's our path.
         String result = classLoader.toString();
-        int index = result.lastIndexOf('[');
-        result = (index == -1) ? result : result.substring(index + 1);
-        index = result.indexOf(']');
-        return (index == -1) ? result : result.substring(0, index);
+
+        // There are differences in Android versions. For old versions, this is just about taking
```

Comment:
Could you do a few things to make this easier to maintain?

1. Pull it out into its own method
2. Add a test case for the old form & new form?
3. Add an example of the toString output here?